gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -663,7 +663,7 @@ public void apply(JobListener jobListener, JobContext jobContext)
   @VisibleForTesting
   public static long sumWorkUnitsSizes (WorkUnitStream workUnitStream) {
     Collection<WorkUnit> workUnits = JobLauncherUtils.flattenWorkUnits(workUnitStream.getMaterializedWorkUnitCollection());
-    long totalSizeInBytes = workUnits.stream().mapToLong(wu -> wu.getPropAsLong(ServiceConfigKeys.WORK_UNIT_SIZE)).sum();
+    long totalSizeInBytes = workUnits.stream().mapToLong(wu -> wu.getPropAsLong(ServiceConfigKeys.WORK_UNIT_SIZE, 1)).sum();
```

Comment:
If it does not have the size, I would assume that size is 0. Any reason to use 1 here?

Reply:
We had a case where a user started a retention job with progress reporting that failed since the Retention Source does not set a value for this property. However, I also noticed that if all the workunits have size 0 and progress reporting is turned on then there will be a divide by 0 error [here](https://jarvis.corp.linkedin.com/codesearch/result/?name=TaskStateCollectorService.java&path=gobblin-elr%2Fgobblin-runtime%2Fsrc%2Fmain%2Fjava%2Forg%2Fapache%2Fgobblin%2Fruntime&reponame=linkedin%2Fgobblin-elr#281). I defaulted to using "1" for the size so progress reporting will default to being based on # work units. Although retention jobs will occur very quickly the progress will appear "100%" instead of "0%" or "Not calculated" if I handled the error within the TaskStateCollectorService.

Reply:
I think we should fix this division by zero where it occurs. It's still possible for the job to copy 0 bytes. For example, when it copies empty files.

In the place where we calculate progress, can we fallback to counting the work units if total size is zero? So the variable says total size in bytes, and it will confuse future developers if we put work unit count there.